File: src/ComposeUI.Messaging.Core/Messages/InvokeResponse.cs
namespace ComposeUI.Messaging.Core.Messages;

public sealed class InvokeResponse : Message
{
    public InvokeResponse(string requestId, string? payload, string? error = null)
    {
        RequestId = requestId;
        Payload = payload;
        Error = error;
    }

    public override MessageType Type => MessageType.InvokeResponse;
    public string RequestId { get; init; }
    public string? Payload { get; init; }
    public string? Error { get; init; }
}

File: Tryouts/Messaging-JS/src/protocol/messages/InvokeResponse.ts
import { MessageBuffer } from "../../MessageBuffer";
import { AbstractResponse } from "./AbstractResponse";
import { Message } from "./Message";

export interface InvokeResponse extends AbstractResponse {
    type: "InvokeResponse";
    payload?: MessageBuffer;
}

export function isInvokeResponse(message: Message): message is InvokeResponse {
    return message.type == "InvokeResponse";
}
